0.17/files/update-mods.sh

```diff
@@ -25,63 +25,75 @@ print_failure()
 
 update_mod()
 {
-  MOD_NAME=$1
+  MOD_NAME="$1"
 
-  print_step "Trying to update $MOD_NAME"
+  print_step "Checking for update of mod $MOD_NAME..."
 
   MOD_INFO_URL="$MOD_BASE_URL/api/mods/$MOD_NAME"
-  MOD_INFO_JSON=$(curl --silent $MOD_INFO_URL)
+  MOD_INFO_JSON=$(curl --silent "$MOD_INFO_URL")
   
-  MOD_INFO=$(curl --silent $MOD_INFO_URL | jq -j --arg version $FACTORIO_VERSION ".releases|reverse|map(select(.info_json.factorio_version as \$mod_version | \$version | startswith(\$mod_version)))[0]|.file_name, \" \", .download_url, \" \", .sha1")
+  MOD_INFO=$(echo "$MOD_INFO_JSON" | jq -j --arg version $FACTORIO_VERSION ".releases|reverse|map(select(.info_json.factorio_version as \$mod_version | \$version | startswith(\$mod_version)))[0]|.file_name, \";\", .download_url, \";\", .sha1")
 
-  set -- $MOD_INFO
-  MOD_FILENAME=$1
-  MOD_URL=$2
-  MOD_SHA1=$3
+  MOD_FILENAME=$(echo "$MOD_INFO" | cut -f1 -d";")
+  MOD_URL=$(echo "$MOD_INFO" | cut -f2 -d";")
+  MOD_SHA1=$(echo "$MOD_INFO" | cut -f3 -d";")
 
   if [ -z "$MOD_URL" ]; then
     return 1
   fi
 
   if [ "$MOD_FILENAME" = "null" ]; then
-    print_failure "  Mod Not compatible with version"
+    print_failure "  Not compatible with version"
     return 1
   fi
 
-  if [ -f $MOD_DIR/$MOD_FILENAME ]; then
-    print_success "  Already up-to-date"
+  if [ -f "$MOD_DIR/$MOD_FILENAME" ]; then
+    print_success "  Already up-to-date."
     return 0
   fi
 
-  FULL_URL=$MOD_BASE_URL$MOD_URL?username=$USERNAME\&token=$TOKEN 
-  HTTP_STATUS=$(curl --silent -L -w "%{http_code}" -o $MOD_DIR/$MOD_FILENAME $FULL_URL)
+  print_step "Downloading..."
+  FULL_URL="$MOD_BASE_URL$MOD_URL?username=$USERNAME&token=$TOKEN"
+  HTTP_STATUS=$(curl --silent -L -w "%{http_code}" -o "$MOD_DIR/$MOD_FILENAME" "$FULL_URL")
+  echo $FULL_URL
 
-  set -- $(sha1sum $MOD_DIR/$MOD_FILENAME)
+  if [ "$HTTP_STATUS" != "200" ]; then
+    print_failure "  Download failed: Code $HTTP_STATUS."
+    rm "$MOD_DIR/$MOD_FILENAME"
+    return 1
+  fi
+
+  if [ ! -f "$MOD_DIR/$MOD_FILENAME" ]; then
+    print_failure "  Downloaded file missing!"
+    return 1
+  fi
+
+  set -- $(sha1sum "$MOD_DIR/$MOD_FILENAME")
```

Comment:
missing quotes

Reply:
No, that doesn't work. When quoting the "sha1sum" result, the returned values are just one argument instead of multiple arguments, breaking the complete SHA checking-part. Currently three arguments are returned: $1 = SHA, $2 = filename. When quoting $1 will be both together.

Reply:
I don't like that approach to much but I think it is fine here.

Reply:
That wasn't done by me - I had no reason to change it. Parsing with `cut` or something else might also work, however IMHO it also works this way so I think it's fine.